### PR TITLE
ci: Staging deploy GH action definition

### PR DESCRIPTION
**Pull request by lavalamp-**
## Description

This PR introduces a new GitHub action that will run off pushes to the `releases/staging` branch. Specifically it:

- Prepares the build environment
- Builds a "staging" version of the `interface` code (presently only code difference is Sentry reporting)
- Deploys the "staging" version of the `interface` code to a staging Cloudflare Page
- Sends the source maps to Sentry

### QA (ie manual testing)

Once all of the new CI PRs are landed we will test the system end-to-end.

### CODE REVIEW

File: src/utils/env.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export function isAppUniswapOrg({ hostname }: { hostname: string }): boolean {
 
 export function isSentryEnabled(): boolean {
   // Disable in e2e test environments
-  if (isStagingEnv() || (isProductionEnv() && !isAppUniswapOrg(window.location))) return false
```
Comment (lavalamp-) [inline src/utils/env.ts:24]: I'm thinking this is necessary... No point in updating Sentry environment to staging if we're not actually going to do any reporting??

Comment (zzmp) [reply src/utils/env.ts:24]: yes, I just wonder where else this is being set. Probably in our vercel builds, which we do _not_ want reporting to sentry. I think we may want an additional check on window.location, but that can be done in a follow-up.

Comment (zzmp) [reply src/utils/env.ts:24]: Could you change this to `(isStagingEnv() && !isAppUniswapStagingOrg(window.location))`? That way we'll only send sentry events from the staging build.

**Review by zzmp — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-to-staging.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+on:
+  push:
+    branches:
+      - 'releases/staging'
+
+jobs:
+  deploy-to-staging:
+    runs-on: ubuntu-latest
+    environment:
+      name: release
```
[the remark this answers is not in the document]
Comment (lavalamp-) [reply .github/workflows/deploy-to-staging.yml:11]: Depends on where we're storing the new Cloudflare credentials! If they're stored in a GH environment titled `staging` then yes sounds good :)

File: src/utils/env.ts (hunk 1 above)
Comment (lavalamp-) [reply src/utils/env.ts:24]: Sounds good :)

File: .github/workflows/deploy-to-staging.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Deploy to staging
```
Comment (zzmp) [inline .github/workflows/deploy-to-staging.yml:1]: rename the file to match other ordinal actions: `2-deploy-to-staging.yml`?

File: .github/workflows/deploy-to-staging.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zzmp) [reply .github/workflows/deploy-to-staging.yml:11]: yes, I'm going to create an environment called staging for this 🤝

**Review by zzmp — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-to-staging.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zzmp) [reply .github/workflows/deploy-to-staging.yml:11]: done

[the remark this answers is not in the document]
Comment (lavalamp-) [reply .github/workflows/deploy-to-staging.yml]: 🙏

File: .github/workflows/deploy-to-staging.yml (hunk 3 above)
Comment (lavalamp-) [reply .github/workflows/deploy-to-staging.yml]: Done!

**Review by zzmp — APPROVED** [review]

**Pull request merged**